test/components/legendTests.ts

```diff
@@ -291,6 +291,15 @@ describe("Legend", () => {
     svg.remove();
   });
 
+  it("do not change the order of legend entries when using default comparator", () => {
+    let colorDomain = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k"];
+    color.domain(colorDomain);
+    legend.renderTo(svg);
+    let entryTexts = svg.selectAll(entrySelector)[0].map((node: Element) => d3.select(node).select("text").text());
```

Comment:
If I use data(), I will get the data that is assigned to that entry, which are not the entity text.